### PR TITLE
merge-styles: allowing css variables to be registered without being kebabcased.

### PR DESCRIPTION
**Pull request by dzearing**
Updating `merge-styles` to allow for css variable registration without kebab casing:

Before, type error and erroneous output:
```tsx
mergeStyles({ '--button-borderColor': 'red' }) // output: .css-0{--button-border-color: 'red'}
```
After, no red squiggles, correct output:

```tsx
mergeStyles({ '--button-borderColor': 'red' }) // output: .css-0{--button-borderColor: 'red'}
```

### CODE REVIEW

File: packages/merge-styles/src/IStyle.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { IRawStyleBase } from './IRawStyleBase';
  * {@docCategory IRawStyle}
  */
 export interface IRawStyle extends IRawStyleBase {
+  [key: string]: string | number | undefined | Record<string, IStyle>;
```
Comment (xugao) [inline packages/merge-styles/src/IStyle.ts:10]: add this in `IRawStyleBase`?

Comment (xugao) [reply packages/merge-styles/src/IStyle.ts:10]: side note, would be nice to have this one day: https://github.com/Microsoft/TypeScript/issues/6579

File: packages/merge-styles/src/transforms/kebabRules.test.ts [hunk 2]
```diff
@@ -16,4 +16,12 @@ describe('kebabRules', () => {
 
     expect(rules).toEqual(['-webkit-font-smoothing', 'antialiased']);
   });
+
+  it('can avoid kebabing css variables', () => {
```
Comment (xugao) [inline packages/merge-styles/src/transforms/kebabRules.test.ts:20]: let's also add a test for 
```
const rules = ['-webkit-font-smoothing', 'antialiased'];
```

File: packages/merge-styles/src/IStyle.ts (hunk 1 above)
Comment (dzearing) [reply packages/merge-styles/src/IStyle.ts:10]: i wish

Comment (dzearing) [reply packages/merge-styles/src/IStyle.ts:10]: hmm. I did this first, but then it put the knowledge of the selectors in `IRawStyleBase`. I *think* this is more correct.

**Pull request merged**